packages/@aws-cdk/aws-servicecatalog/test/portfolio.test.ts

```diff
@@ -256,6 +256,131 @@ describe('Portfolio', () => {
       // If this were not idempotent, the second call would produce an error for duplicate construct ID.
       portfolio.giveAccessToRole(role);
       portfolio.giveAccessToRole(role);
+    }),
+
+    test('add tag options to portfolio', () => {
+      const tagOptions = {
+        key1: [
+          { value: 'value1' },
+          { value: 'value2' },
+        ],
+        key2: [{ value: 'value1' }],
+      };
+
+      portfolio.addTagOptions(tagOptions);
+
+      expect(stack).toCountResources('AWS::ServiceCatalog::TagOption', 3); //Generates a resource for each key-value pair
+      expect(stack).toHaveResource('AWS::ServiceCatalog::TagOptionAssociation');
+    }),
+
+    test('fails to add tag options with invalid key length', () => {
+      const tagOptions = {
+        key0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000: [
```

Comment:
Use the `String.repeat()` method instead.

Reply:
took me a while to figure out how to get a var substitution in here, but can accomplish this with using brackets ```['key1'.repeat(1000)]:```